clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecords.java

```diff
@@ -55,7 +55,7 @@ public ConsumerRecords(Map<TopicPartition, List<ConsumerRecord<K, V>>> records)
             throw new IllegalArgumentException("Topic must be non-null.");
         List<List<ConsumerRecord<K, V>>> recs = new ArrayList<List<ConsumerRecord<K, V>>>();
         for (Map.Entry<TopicPartition, List<ConsumerRecord<K, V>>> entry : records.entrySet()) {
-            if (entry.getKey().equals(topic))
+            if (entry.getKey().topic().equals(topic))
```

Comment:
Intellij helped me. :) Once we eliminate all the existing issues, I would like to configure `abide` as part of the build (https://github.com/scala/scala-abide) so that they are prevented in the future.